packages/tools/kda-cli/src/questions/devnet.ts

```diff
@@ -0,0 +1,247 @@
+import { spawned } from '../utils/spawn.js';
+
+import { IAnswers, IQuestion } from './questions.js';
+
+import { readFileSync, writeFileSync } from 'fs';
+import { parse, stringify } from 'yaml';
+
+type StringReducer = (str: string) => string;
+
+const compose =
+  (...reducers: StringReducer[]): StringReducer =>
+  (env: string) =>
+    reducers.reduce((res, reducer) => reducer(res), env);
+
+const replaceValue =
+  (key: string, value: string | number): StringReducer =>
+  (env: string) =>
+    // eslint-disable-next-line @rushstack/security/no-unsafe-regexp
+    env.replace(new RegExp(`(${key}= ?).*`), `$1${value}`);
+
+const composeEnv = (servicePort: number, stratumPort: number): StringReducer =>
+  compose(
+    replaceValue(
+      'CHAINWEB_NODE_IMAGE',
+      'ghcr.io/kadena-io/chainweb-node:sha-7e11817',
+    ),
+    replaceValue('HOST_SERVICE_PORT', servicePort),
+    replaceValue('HOST_STRATUM_PORT', stratumPort),
+  );
+
+const getEnvL1: StringReducer = composeEnv(8080, 1917);
+const getEnvL2: StringReducer = composeEnv(8081, 1918);
+
+const setupEnvFor = async (layer: 'l1' | 'l2'): Promise<void> => {
+  const envFile = `${process.env.HOME}/.devnet/${layer}/.env`;
+  const env = readFileSync(envFile, 'utf-8');
+  if (layer === 'l2') return writeFileSync(envFile, getEnvL2(env), 'utf-8');
+  return writeFileSync(envFile, getEnvL1(env), 'utf-8');
+};
+
+const getDockerFile = (env: 'l1' | 'l2') => {
+  if (env === 'l2')
+    return `${process.env.HOME}/.devnet/${env}/docker-compose.minimal-l2.yaml`;
+  return `${process.env.HOME}/.devnet/${env}/docker-compose.minimal.yaml`;
```

Comment:
Can we create a util that identifies all env vars in one place, does parsing where necessary and has a type for them?

Akin: https://github.com/kadena-community/kadena.js/blob/master/packages/apps/graph/src/utils/dotenv.ts